src/org/jitsi/impl/neomedia/transform/RetransmissionRequesterImpl.java

```diff
@@ -15,108 +15,46 @@
  */
 package org.jitsi.impl.neomedia.transform;
 
-import java.io.*;
-import java.util.*;
-
-import org.jitsi.impl.neomedia.*;
-import org.jitsi.impl.neomedia.rtcp.*;
-import org.jitsi.impl.neomedia.rtp.*;
 import org.jitsi.service.neomedia.*;
-import org.jitsi.service.neomedia.codec.*;
-import org.jitsi.service.neomedia.format.*;
 import org.jitsi.util.*;
+import org.jitsi.util.concurrent.*;
 
 /**
- * Detects lost RTP packets for a particular <tt>RtpChannel</tt> and requests
- * their retransmission by sending RTCP NACK packets.
+ * Creates classes to handle both the detection of loss and the creation
+ * and sending of nack packets, and a scheduler to allow for nacks to be
+ * re-transmitted at a set interval
  *
- * @author Boris Grozev
- * @author George Politis
+ * @author bbaldino
  */
 public class RetransmissionRequesterImpl
     extends SinglePacketTransformerAdapter
     implements TransformEngine, RetransmissionRequester
 {
-    /**
-     * If more than <tt>MAX_MISSING</tt> consecutive packets are lost, we will
-     * not request retransmissions for them, but reset our state instead.
-     */
-    private static final int MAX_MISSING = 100;
-
-    /**
-     * The maximum number of retransmission requests to be sent for a single
-     * RTP packet.
-     */
-    private static final int MAX_REQUESTS = 10;
-
-    /**
-     * The interval after which another retransmission request will be sent
-     * for a packet, unless it arrives. Ideally this should not be a constant,
-     * but should be based on the RTT to the endpoint.
-     */
-    private static final int RE_REQUEST_AFTER = 150;
-
-    /**
-     * The <tt>Logger</tt> used by the <tt>RetransmissionRequesterImpl</tt> class
-     * and its instances to print debug information.
-     */
-    private static final Logger logger
-        = Logger.getLogger(RetransmissionRequesterImpl.class);
-
-    /**
-     * Maps an SSRC to the <tt>Requester</tt> instance corresponding to it.
-     * TODO: purge these somehow (RTCP BYE? Timeout?)
-     */
-    private final Map<Long, Requester> requesters = new HashMap<>();
-
     /**
      * Whether this {@link RetransmissionRequester} is enabled or not.
      */
     private boolean enabled = true;
 
-    /**
-     * The thread which requests retransmissions by sending RTCP NACK packets.
-     */
-    private final Thread thread;
-
     /**
      * Whether this <tt>PacketTransformer</tt> has been closed.
      */
     private boolean closed = false;
 
-    /**
-     * The {@link MediaStream} that this instance belongs to.
-     */
-    private final MediaStream stream;
+    protected final RetransmissionRequesterDelegate retransmissionRequesterDelegate;
 
-    /**
-     * The SSRC which will be used as Packet Sender SSRC in NACK packets sent
-     * by this {@code RetransmissionRequesterImpl}.
-     */
-    private long senderSsrc = -1;
+    RecurringRunnableExecutor recurringRunnableExecutor = new RecurringRunnableExecutor();
```

Comment:
Is there anything stopping us from making this static and saving some threads?

Reply:
good idea! yeah i think it seems reasonable that one thread could handle all the nacks, any thoughts on if heavy load might cause a problem there? worst case we could be missing targets for nack transmission.

Reply:
I guess heavy load could be a problem, but I doubt that it will be a bottleneck (and I would imagine it would also be a problem for the multiple threads scenario, too).